pinax/stripe/actions/invoices.py

```diff
@@ -63,9 +63,116 @@ def pay(invoice, send_receipt=True):
     return False
 
 
+def paid(invoice):
+    """
+    Sometimes customers may want to pay with payment methods outside of Stripe, such as check.
+    In these situations, Stripe still allows you to keep track of the payment status of your invoices.
+    Once you receive an invoice payment from a customer outside of Stripe, you can manually
+    mark their invoices as paid.
+
+    Args:
+        invoice: the invoice object to close
+    """
+    if not invoice.paid:
+        stripe_invoice = invoice.stripe_invoice
+        stripe_invoice.paid = True
+        stripe_invoice_ = stripe_invoice.save()
+        sync_invoice_from_stripe_data(stripe_invoice_)
+
+
+def forgive(invoice):
+    """
+    Forgiving an invoice instructs us to update the subscription status as if the invoice were
+    successfully paid. Once an invoice has been forgiven, it cannot be unforgiven or reopened.
+
+    Args:
+        invoice: the invoice object to close
+    """
+    if not invoice.paid:
+        stripe_invoice = invoice.stripe_invoice
+        stripe_invoice.forgiven = True
+        stripe_invoice_ = stripe_invoice.save()
+        sync_invoice_from_stripe_data(stripe_invoice_)
+
+
+def close(invoice):
+    """
+    Cause an invoice to be closed; This prevents Stripe from automatically charging your customer for the invoice amount.
+
+    Args:
+        invoice: the invoice object to close
+    """
+    if not invoice.closed:
+        stripe_invoice = invoice.stripe_invoice
+        stripe_invoice.closed = True
+        stripe_invoice_ = stripe_invoice.save()
```

Comment:
Again, let's just call `save()` and pass the original object to the sync method.